ts/src/index.ts

```diff
@@ -129,13 +155,15 @@ export async function start(config: Config = {}): Promise<void> {
         config);
     return;
   }
+
   let normalizedConfig: ProfilerConfig;
```

Comment:
Looking more into it now, it feels you could move all of
```
let normalizedConfig: ProfilerConfig;
...
profiler = new Profiler(normalizedConfig);
```
into `initConfigAndMaybeStartHeapProfiler` naming it as createProfile() which returns the created profiler or null and call it as "profiler = createProfiler(); if (!profiler) return;" (the logging would be inside of createProfiler).

Reply:
Done.